### PR TITLE
improve code clarity

### PR DESCRIPTION
**Pull request by SimHoZebs**
See comments for individual improvements.

### CODE REVIEW

File: functions/boardWithRandomlyPlacedShips .ts [hunk 1]
```diff
@@ -1,18 +1,13 @@
 /**Places ships from randomly generated coordinates that is not outside the grid or already occupied on any provided 10*10 grid*/
 const boardWithRandomlyPlacedShips = (
   shipLength: number,
-  Board: number[][],
+  board: number[][],
```
Comment (SimHoZebs) [inline functions/boardWithRandomlyPlacedShips .ts:4]: every variable and constant should start uncapitalized

File: functions/boardWithRandomlyPlacedShips .ts [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
 ): number[][] => {
   const width = 10;
   const orientation =
     Math.round(Math.random()) >= 0.5 ? "horizontal" : "vertical";
-  console.log("running");
 
-
-  const updatedBoard = [...Board];
```
Comment (SimHoZebs) [inline functions/boardWithRandomlyPlacedShips .ts:12]: This is the same as using `board`, as spread operator copies items by reference.

File: interfaces/types.ts [hunk 3]
```diff
@@ -7,11 +7,7 @@ export interface IFinishedProps {
   restart: () => void;
 }
 
-export type Ship = {
-  type: ShipNames;
-};
-
-export enum ShipNames {
```
Comment (SimHoZebs) [inline interfaces/types.ts:14]: This represents ship lengths based on their names, not the other way around.

File: stage/Game.tsx [hunk 4]
```diff
@@ -1,48 +1,39 @@
...
-      )
+    setOpponentBoard(() => newBoard());
+    shipLengths.map((length) =>
+      setOpponentBoard((prev) => boardWithRandomlyPlacedShips(length, prev))
     );
 
     console.table(opponentBoard);
   }
 
-  useEffect(() => {
```
Comment (SimHoZebs) [inline stage/Game.tsx:34]: useEffect is pointless if these values are assigned to the state on init.

File: stage/Game.tsx [hunk 5]
```diff
@@ -1,48 +1,39 @@
...
-    ShipNames.Cruiser,
-  ];
+  function newBoard() {
+    return Array(width)
+      .fill(0)
+      .map(() => Array(width).fill(0));
+  }
 
-  const width = 10;
+  const shipLengths = [
```
Comment (SimHoZebs) [inline stage/Game.tsx:19]: they are array of ships lengths :D

**Review by ipod1g — APPROVED** [review]
Just a small question, isn't using board argument straight into our function operations not desirable due to mutability? Can you perhaps explain a bit on this?

**Comment by SimHoZebs** [comment]
> Just a small question, isn't using board argument straight into our function operations not desirable due to mutability? Can you perhaps explain a bit on this?

`board` can't be mutated by changing `board`. It only changes from using `setBoard`.

**Pull request merged**